test/http2_test/test_rst_after_data.py

```diff
@@ -0,0 +1,28 @@
+import http2_base_server
+
+class TestcaseRstStreamAfterData(object):
+  """
+    In response to an incoming request, this test sends headers, followed by
+    data, followed by a reset stream frame. Client asserts that the RPC failed.
+    Client needs to deliver the complete message to the application layer.
```

Comment:
Are we sure that all client languages can do this? For instance,Java can only deliver the completed message to the application layer using the async API. I think PHP, at least, might still be missing async calls.